mllib/src/main/scala/org/apache/spark/mllib/linalg/BLAS.scala

```diff
@@ -317,19 +317,23 @@ private[spark] object BLAS extends Serializable with Logging {
       B: DenseMatrix,
       beta: Double,
       C: DenseMatrix): Unit = {
+    val transModA = transA ^ A.isTransposed
```

Comment:
If we set the transpose flag in matrix, maybe we should remove the `transA` and `transB` arguments  from the BLAS interface: `gemm(alpha, A, B, beta, C)`.